### PR TITLE
feat: [IDP-3835]: Added the release notes for v0.34.0

### PR DESCRIPTION
**Pull request by Debanitrkl**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: release-notes/internal-developer-portal.md [hunk 1]
```diff
@@ -17,6 +17,26 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
 
+## October 2024
+
+### Version 0.34.0
+
+<!-- October 1, 2024 -->
+
+#### New features and enhancements
+
+- We have added the new Workflows Homepage, which helps you to customize the Workfows organization and information associated. [IDP-3752]
```
Comment (sandeepa-kv-jois) [inline release-notes/internal-developer-portal.md:28]: Let's also mention about FF IDP_ENABLE_WORKFLOWSV2 requirement for this

File: release-notes/internal-developer-portal.md [hunk 2]
```diff
@@ -17,6 +17,26 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
+
+<!-- October 1, 2024 -->
+
+#### New features and enhancements
+
+- We have added the new Workflows Homepage, which helps you to customize the Workfows organization and information associated. [IDP-3752] 
+- We have upgraded to the new backend system of Backstage in Harness IDP. [IDP-3252]
+- Added Support for usage of pipeline templates in the pipeline orchestrator used in workflows. [IDP-3599]
+- Enhanced the API Response for Catalog Ingestion API. [IDP-3672]
+
```
Comment (sandeepa-kv-jois) [inline release-notes/internal-developer-portal.md:32]: + integrated Sysdig plugin

File: release-notes/internal-developer-portal.md [hunk 3]
```diff
@@ -17,6 +17,28 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
 
+## October 2024
+
+### Version 0.34.0
+
+<!-- October 1, 2024 -->
+
+#### New features and enhancements
+
+- We have added the new Workflows Homepage, which helps you to customize the Workflows organization and information associated. This feature is behind the Feature Flag `IDP_ENABLE_WORKFLOWSV2`, contact [harness support]((mailto:support@harness.io)) to enable it on your account. [IDP-3752]
```
Comment (OrkoHunter) [inline release-notes/internal-developer-portal.md:28]: add link the the page we just created

File: release-notes/internal-developer-portal.md [hunk 4]
```diff
@@ -17,6 +17,28 @@ Review the notes below for details about recent changes to Harness Internal Deve
...
+- We have upgraded to the new backend system of Backstage in Harness IDP. [IDP-3252]
+- Added Support for usage of pipeline templates in the pipeline orchestrator used in workflows. [IDP-3599]
+- Enhanced the API Response for Catalog Ingestion API. [IDP-3672]
+- New plugins added to the marketplace.
+  - [Sysdig Plugin for Backstage](https://github.com/sysdiglabs/backstage-plugin-sysdig)
+
+#### Bug fixes
+
+- Fixed the issue with `mode: append` to support updates to array values, in Catalog Ingestion API. [IDP-3734]
+- Fixed the issue with Sonrqube Plugin. [IDP-3691]
```
Comment (OrkoHunter) [inline release-notes/internal-developer-portal.md:38]:
```suggestion
- Fixed the issue with Sonarqube Plugin. [IDP-3691]
```

What was the issue? Should the user care?

**Review by sandeepa-kv-jois — CHANGES_REQUESTED** [review]
+ Catalog API fix with x-api-key in header

**Review by OrkoHunter — APPROVED** [review]

**Pull request merged**